lua/nvim-tree/actions/tree/modifiers/expand.lua

```diff
@@ -72,12 +129,13 @@ local function gen_iterator()
 end
```

Comment:
The max folder discovery test above does not appear to be necessary as `limit_folder_discovery` does the job.

Commenting this out results in the correct number of folders being expanded, just no message.

Having the same test twice in two locations is fragile and unclear.

Proposal: `limit_folder_discovery` prints the warning and `gen_iterator`'s function returns nothing.